### PR TITLE
add string word and add error N1QL query syntax

### PR DESCRIPTION
**Pull request by giallon**
now we can  write query N1QL as string.

Bonus : I had error N1QL query syntax.

### CODE REVIEW

File: lib/libcouchbase/connection.rb [hunk 1]
```diff
@@ -867,16 +866,27 @@ def query_callback_common(row_data)
...
+                    view.error error_klass.new(body_text)
                 else
                     view.error error_klass.new
                 end
             end
         end
 
+        # Extracts the row content of a response
+        def row_text(row_data)
+            if (row_data[:nrow]).positive?
```
Comment (fab-girard) [inline lib/libcouchbase/connection.rb]: remove bracket

File: lib/libcouchbase/connection.rb [hunk 2]
```diff
@@ -16,7 +16,7 @@
             rescue => e
             end
         end
-        sleep 2 if connections.length > 0
```
Comment (fab-girard) [inline lib/libcouchbase/connection.rb]: keep non numeric predicate please

File: lib/libcouchbase/connection.rb [hunk 3]
```diff
@@ -867,16 +866,27 @@ def query_callback_common(row_data)
...
+                row_data[:row].read_string(row_data[:nrow])
+            else
+                ''
+            end
+        end
+
         # Extracts the body content of a HTTP response
         def body_text(http_resp)
-            if http_resp[:nbody] > 0
+            if (http_resp[:nbody]).positive?
```
Comment (fab-girard) [inline lib/libcouchbase/connection.rb]: remove bracket and keep non numeric predicate

File: lib/libcouchbase/n1ql.rb [hunk 4]
```diff
@@ -23,6 +23,14 @@ def initialize(bucket, explain: false, **options)
         attr_accessor *Ordering
         attr_accessor :explain
         attr_reader   :bucket, :connection
+        attr_accessor :string
+
+        def string(val = nil)
```
Comment (fab-girard) [inline lib/libcouchbase/n1ql.rb]: can you explain me where do you need this method and in what it need? it's strange to return sometimes a string, sometimes an object...

Comment (giallon) [reply lib/libcouchbase/n1ql.rb]: the goal is to write query like it
```ruby
bucket.n1ql.query("
          SELECT city, COUNT(*) AS cnt FROM `travel-sample`
          WHERE type = $type
          GROUP BY city
          ORDER BY cnt DESC
          LIMIT 3"
```

**Pull request merged**